includes/admin/settings/class-wc-settings-products.php

```diff
@@ -304,6 +313,7 @@ public function get_settings( $current_section = '' ) {
 							'id'       => 'woocommerce_shop_page_id',
 							'type'     => 'single_select_page',
 							'default'  => '',
+							'args'     => array( 'exclude' => $draft_pages ),
```

Comment:
Drafts are included using `post_type` in https://github.com/woocommerce/woocommerce/blob/3377c60c5e5602a4c82b6ef43b7b48b9c895f9ad/includes/admin/class-wc-admin-settings.php#L563

So should be easier to just include the types you like instead of querying all drafts and excluding it. 

```suggestion
							'args'     => array( 'post_status' => 'publish,private' ),
```

Reply:
@claudiosanches 

Makes sense. Will fix it today.